src/components/Products/Products.js

```diff
@@ -0,0 +1,76 @@
+import React, { useEffect, useState } from "react";
+import { useProductList } from "../../context/ProductContext/ProductContext";
+import { Filters } from "../Filters/Filters";
+import { Toast } from "../Toast/Toast";
+import { Link } from "@reach/router";
+import Loader from "react-loader-spinner";
+
+const ProductsPage = () => {
+  const { productList, loading } = useProductList();
+  let [toastMessageList, setToastMessageList] = useState([]);
```

Comment:
You can make a custom hook `useToast` which shows the toast and returns a method `setToast` to updates messages
Try it out, or message me we will do it live on VC!